lib/events.js

```diff
@@ -203,37 +200,32 @@ EventEmitter.prototype.emit = function emit(type) {
     return false;
   }
 
-  handler = events[type];
+  const handler = events[type];
 
-  if (!handler)
+  if (handler === undefined)
     return false;
 
-  if (domain && this !== process) {
+  let needDomainExit = false;
+  if (domain !== null && domain !== undefined && this !== process) {
     domain.enter();
     needDomainExit = true;
   }
 
-  var isFn = typeof handler === 'function';
-  len = arguments.length;
-  switch (len) {
-    // fast cases
-    case 1:
+  const isFn = typeof handler === 'function';
+  switch (args.length) {
```

Comment:
Wait if we're using rest args, while the unwinding?

Reply:
It's still faster to have these unwound cases than just using `.apply(target, args)`. The main speed up is that we no longer need to create a new array in the slow case and manually copy arguments. (The fast cases are also slightly faster when using the `args` than `arguments`.)

Reply:
can you gist those permutations?
I'm writing a benchmark so that we can track V8 optimizations.

BTW: I'm also seeing faster results for unwound calls, but <10% differance

Reply:
Yeah, the difference isn't massive but I'm not comfortable having a performance regression. Will put together a gist this weekend if I can recall exactly what it was I tested.